x-pack/solutions/security/plugins/elastic_assistant/server/routes/attack_discovery/helpers/generate_and_update_discoveries.ts

```diff
@@ -90,26 +91,43 @@ export const generateAndUpdateAttackDiscoveries = async ({
       telemetry,
     });
 
+    let storedAttackDiscoveries = attackDiscoveries;
     if (attackDiscoveryAlertsEnabled) {
       const alertsContextCount = anonymizedAlerts.length;
 
+      // Deduplicate attackDiscoveries before creating alerts
+      const indexPattern = dataClient.getScheduledAndAdHocIndexPattern();
```

Comment:
Let's use ad hoc index pattern only, since owner ID will make sure alerts never overlap in scheduled and ad hoc indices. cc @andrew-goldstein

Reply:
Done in https://github.com/elastic/kibana/pull/223483/commits/6ff843835247c145ea63583057a9da85493b6e5a